src/main/java/ws/wamp/jawampa/WampClient.java

```diff
@@ -367,6 +378,13 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
                         featuresNode.put("pattern_based_subscription", true);
                     }
                 }
+                if(authId != null) {
+                    o.put("authid", authId);
+                }
+                ArrayNode authMethodsNode = o.putArray("authmethods");
```

Comment:
Shouldn't this only written when authentication was configured to avoid sending extra fields for standard connections?
It could be either put in the if block for authId or in an extra check whether `authMethods` is not null and not empty.